rate-limiting-global/src/main/java/io/stargate/db/limiter/global/impl/GlobalRateLimitingManager.java

```diff
@@ -0,0 +1,143 @@
+import io.stargate.db.limiter.AsyncRateLimiter;
+import io.stargate.db.limiter.RateLimitingDecision;
+import io.stargate.db.limiter.RateLimitingManager;
+import io.stargate.db.limiter.RateLimitingManager.ConnectionManager;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * A rate limiting manager that rate limit all queries globally (except reads on system tables) to a
+ * pre-configured amount of queries per seconds.
+ *
+ * <p>The enforced QPS is defined through the {@link #RATE_PROPERTY} system property.
+ *
+ * <p>As of this writing, this limiter serves mostly for testing (see the {@code
+ * GlobalRateLimitingTest}) and as a demonstration of how to write a simple {@link
+ * RateLimitingManager}. However, it might need some improvements to be truly usable in production.
+ * Mainly, the rate cannot be currently configured at runtime, which is limiting (pun intended).
+ * Additionally, this avoid (by default) limiting some reads on system tables, but the way this is
+ * done is fragile/inflexible.
+ */
+public class GlobalRateLimitingManager implements RateLimitingManager, ConnectionManager {
+  /*
+   * TODO: This
+   *
+   * Currently, the rate is set by a system property at startup and can't change, which makes
+   *  this not that usable in real life. Of course, the fact that it is a pretty big hammer also
+   *  limits the usefulness. So this is currently more for 1) testing rate limiting and 2)
+   *  demonstrates how one can write a `RateLimitingManager`.
+   *  But at least, making it possible to change the rate at runtime would be useful (though "how"
+   *  remains a question to be answered).
+   */
+
+  public static final String RATE_PROPERTY = "stargate.limiter.global.rate_qps";
+  private static final int DELAYED_TASKS_EXECUTOR_THREADS =
+      Integer.getInteger("stargate.limiter.global.threads", 4);
+  private static final boolean RATE_LIMIT_SYSTEM_TABLES =
+      Boolean.getBoolean("stargate.limiter.global.enabled_on_system_tables");
+
+  private final AsyncRateLimiter limiter;
+
+  public GlobalRateLimitingManager() {
+    this(buildLimiter(defaultExecutor()));
+  }
+
+  public GlobalRateLimitingManager(AsyncRateLimiter limiter) {
+    this.limiter = limiter;
+  }
+
+  private static ScheduledExecutorService defaultExecutor() {
+    return Executors.newScheduledThreadPool(DELAYED_TASKS_EXECUTOR_THREADS);
+  }
+
+  private static AsyncRateLimiter buildLimiter(ScheduledExecutorService executor) {
+    String rateStr = System.getProperty(RATE_PROPERTY);
+    if (rateStr == null || rateStr.isEmpty()) {
+      throw new IllegalArgumentException(
+          format(
+              "Global rate limiting is enabled but missing (or empty) value for property '%s'",
+              RATE_PROPERTY));
+    }
+
+    try {
+      return new AsyncRateLimiter(
+          executor, Long.parseLong(rateStr), TimeUnit.SECONDS, 1, TimeUnit.MINUTES);
```

Comment:
any reason why we use SECOND and MINUTES? Wouldn't it be more consistent to use the same time-unit for both? It may be more important if both values will be provided via an external setting.

Reply:
Not reason, but I don't see how this is in any way confusing and what "consistency" would add. And this can be changed later easily enough if the context changes.